mutiny-core/src/lib.rs

```diff
@@ -1066,7 +1070,13 @@ impl<S: MutinyStorage> MutinyWalletBuilder<S> {
         mw.check_blind_tokens();
 
         // start the hermes background process
-        mw.start_hermes()?;
+        // get profile key if we have it, we need this to decrypt private zaps
+        let profile_key = match &mw.nostr.primary_key {
+            NostrSigner::Keys(keys) => Some(keys.clone()),
+            #[cfg(target_arch = "wasm32")]
+            NostrSigner::NIP07(_) => None,
+        };
+        mw.start_hermes(profile_key)?;
```

Comment:
We shouldn't be using the user's nostr key for this.

Reply:
It's for decrypting private zaps, not for the actual dms